crates/services/parallel-executor/src/tests/tests_executor.rs

```diff
@@ -0,0 +1,637 @@
+                ))
+                .unwrap();
+
+            // Request for thread 2
+            let (pool_request_params, response_sender) =
+                tx_pool_requests_receiver.recv().unwrap();
+            assert_eq!(pool_request_params.filter.excluded_contract_ids.len(), 1);
+            assert_eq!(
+                pool_request_params
+                    .filter
+                    .excluded_contract_ids
+                    .get(&contract_id_1),
+                Some(&contract_id_1)
+            );
+            response_sender
+                .send((
+                    vec![MaybeCheckedTransaction::CheckedTransaction(
+                        tx_contract_2
+                            .clone()
+                            .into_checked_basic(
+                                0u32.into(),
+                                &ConsensusParameters::default(),
+                            )
+                            .unwrap()
+                            .into(),
+                        0,
+                    )],
+                    TransactionFiltered::NotFiltered,
+                ))
+                .unwrap();
+            // Request for thread 1 again
+            let (pool_request_params, response_sender) =
+                tx_pool_requests_receiver.recv().unwrap();
+            assert_eq!(pool_request_params.filter.excluded_contract_ids.len(), 1);
+            assert_eq!(
+                pool_request_params
+                    .filter
+                    .excluded_contract_ids
+                    .get(&contract_id_2),
+                Some(&contract_id_2)
+            );
+            response_sender
+                .send((vec![], TransactionFiltered::Filtered))
+                .unwrap();
+            // Request for thread 1 or 2 again
+            let (pool_request_params, response_sender) =
+                tx_pool_requests_receiver.recv().unwrap();
+            // TODO: Maybe it's ConsensusParameters::default().block_gas_limit() / number_of_cores
+            assert!(
+                pool_request_params.gas_limit
+                    > ConsensusParameters::default().block_gas_limit() - max_gas
+            );
+            assert_eq!(pool_request_params.filter.excluded_contract_ids.len(), 0);
+            response_sender
+                .send((
+                    vec![MaybeCheckedTransaction::CheckedTransaction(
+                        tx_both_contracts
+                            .clone()
+                            .into_checked_basic(
+                                0u32.into(),
+                                &ConsensusParameters::default(),
+                            )
+                            .unwrap()
+                            .into(),
+                        0,
+                    )],
+                    TransactionFiltered::NotFiltered,
+                ))
+                .unwrap();
+        }
+    });
+}
+
+#[test]
+#[ignore]
+fn execute__utxo_ordering_kept() {
```

Comment:
Interesting. Is it an important property we rely on right now that dependent transactions occur in-order in blocks?